src/cleanup_rules/swift/rules.toml

```diff
@@ -131,6 +133,49 @@ query = """(
 groups = ["boolean_expression_simplify"]
 replace_node = "disjunction_expression"
 replace = "@lhs"
+#
+# Before 
+#   false && abcd()
+# After 
+#   false
+#
+[[rules]]
+name = "false_and_something"
+query = """(
+(conjunction_expression
+        lhs: [(boolean_literal) @false  
+            (tuple_expression 
+                value: (boolean_literal) @false)]
+        rhs: (_) 
+    ) @conjunction_expression
+(#eq? @false "false")
+)"""
+groups = ["boolean_expression_simplify"]
+replace_node = "conjunction_expression"
+replace = "false"
+
+#
+# Before 
+#   abcd && false
+# After 
+#   false
+#
+# f1() && false wouldn't be simplified as f1() can have some internal processing which we don't want to clean up
```

Comment:
```suggestion
# Note that this rule only applies when the lhs of the conjunction is an identifier. 
# In other scenarios (i.e. is a method invocation or some other expression) we would not simplify at all, because such expressions could have side effects
```